desktop/core/src/desktop/js/reactComponents/TaskBrowser/TaskBrowser.tsx

```diff
@@ -0,0 +1,434 @@
+  const [scheduledTasks, setScheduledTasks] = useState([]); // Store an array of scheduled tasks
+
+  const [showLogPopup, setShowLogPopup] = useState(false); // This should match where you use it
+  const [currentTaskId, setCurrentTaskId] = useState(null);
+  const [taskLogs, setTaskLogs] = useState("");
+
+  const handleScheduleSubmit = (taskName, taskParams) => {
+    const payload = {
+      taskName,
+      taskParams,
+    };
+
+    axios.post('/desktop/api2/handle_submit', payload)
+    .catch(error => {
+      console.error('Error scheduling task', error);
+    });
+  };
+
+  // This function will open the schedule task popup
+  const handleSchedulePopup = () => {
+    setShowSchedulePopup(true);
+  };
+
+  // Add the function to handle showing logs
+  const handleShowLogs = (taskId) => {
+    // This function should set the current task ID and fetch the logs
+    console.log("hi im in handlelogs");
+    setCurrentTaskId(taskId);
+    // Fetch the logs from the server
+    axios.get(`/desktop/api2/get_task_logs/${taskId}/`)
+      .then(response => {
+        // Assuming the server response contains the logs in the data object
+        setTaskLogs(response.data);
+        setShowLogPopup(true); // Open the popup
+      })
+      .catch(error => {
+        console.error('Error fetching logs:', error);
+        // Handle the error appropriately
+      });
+  };
+
+  return (
+    <div>
+      {showSchedulePopup && (
+        <ScheduleTaskPopup
+          onClose={() => setShowSchedulePopup(false)}
+          onSubmit={handleScheduleSubmit}
+          open={showSchedulePopup}
+        />
+      )}
+      <div style={{ margin: '20px 0' }}>
+        <TaskBrowserTable tasks={scheduledTasks} handleShowLogs={handleShowLogs} handleSchedulePopup={handleSchedulePopup}/>
+      </div>
+      {showLogPopup && (
+        <div className="popupStyle">
+          <div className="popupContentStyle">
+            <h2>Task Logs - {currentTaskId}</h2>
+            <pre>{taskLogs}</pre>
+            <button onClick={() => setShowLogPopup(false)}>Close</button>
+          </div>
+        </div>
+      )}
+    </div>
+
+  );
+};
+
+
+// A helper function to calculate duration
+const calculateDuration = (start, end) => {
+  if (!start || !end) return 'N/A'; // If either start or end is missing
+
+  const startDate = new Date(start);
+  const endDate = new Date(end);
+  const duration = endDate - startDate; // Duration in milliseconds
+
+  if (isNaN(duration)) return 'Invalid Dates'; // If dates can't be parsed
+
+  // Convert duration to hours, minutes, and seconds
+  let seconds = Math.floor((duration / 1000) % 60);
+  let minutes = Math.floor((duration / (1000 * 60)) % 60);
+  let hours = Math.floor((duration / (1000 * 60 * 60)) % 24);
+
+  // Build duration string
+  let durationStr = '';
+  if (hours > 0) durationStr += hours + 'h, ';
+  if (minutes > 0 || hours > 0) durationStr += minutes + 'm, ';
+  durationStr += seconds + 's';
+
+  return durationStr;
```

Comment:
This seems like a generic util function and should go under `utils/`. That being said we're likey doing this elsewhere in Hue as well so good if consistent. I believe we also depend on moment js that likely has some nice formatter to use.

Reply:
Couldn't find a similar implementation in react, so moved this code to "desktop/core/src/desktop/js/utils/dateTimeUtils.ts"